services/datamanager/builtin/sync/upload_data_capture_file.go

```diff
@@ -41,63 +51,105 @@ func uploadDataCaptureFile(ctx context.Context, f *data.CaptureFile, conn cloudC
 		return 0, nil
 	}
 
-	if md.GetType() == v1.DataType_DATA_TYPE_BINARY_SENSOR && len(sensorData) > 1 {
-		return 0, fmt.Errorf("binary sensor data file with more than one sensor reading is not supported: %s", f.GetPath())
+	sensorDataTypeSet := captureTypesInSensorData(sensorData)
+	if len(sensorDataTypeSet) != 1 {
+		return 0, errMultipleReadingTypes
 	}
 
-	// camera.GetImages is a special case. For that API we make 2 binary data upload requests
-	if md.GetType() == v1.DataType_DATA_TYPE_BINARY_SENSOR && md.GetMethodName() == data.GetImages {
-		logger.Debugf("attemping to upload camera.GetImages data: %s", f.GetPath())
+	_, isTabular := sensorDataTypeSet[data.CaptureTypeTabular]
+	if isLegacyGetImagesCaptureFile(md, isTabular) {
+		logger.Debugf("attemping to upload legacy camera.GetImages data: %s", f.GetPath())
+		return uint64(f.Size()), legacyUploadGetImages(ctx, conn, md, sensorData[0], f.Size(), f.GetPath(), logger)
+	}
 
-		return uint64(f.Size()), uploadGetImages(ctx, conn, md, sensorData[0], f.Size(), f.GetPath(), logger)
+	if err := checkUploadMetadaTypeMatchesSensorDataType(md, sensorDataTypeSet); err != nil {
+		return 0, err
 	}
 
-	metaData := uploadMetadata(conn.partID, md, md.GetFileExtension())
+	metaData := uploadMetadata(conn.partID, md)
 	return uint64(f.Size()), uploadSensorData(ctx, conn.client, metaData, sensorData, f.Size(), f.GetPath(), logger)
 }
 
-func uploadMetadata(partID string, md *v1.DataCaptureMetadata, fileextension string) *v1.UploadMetadata {
-	return &v1.UploadMetadata{
+func checkUploadMetadaTypeMatchesSensorDataType(md *datasyncPB.DataCaptureMetadata, sensorDataTypeSet map[data.CaptureType]struct{}) error {
+	var captureType data.CaptureType
+	// get first element of single element set
+	for k := range sensorDataTypeSet {
```

Comment:
Someone in sol eng or external to viam writes a program that writes capture files and they put both binary and tabular data in that file. That or we have a bad bug.

This is not an expected condition but we are running this check so that if it does happen, we move the file to the failed directory and move on.